buildSrc/src/test/groovy/org/elasticsearch/gradle/VersionCollectionTests.groovy

```diff
@@ -16,6 +20,7 @@ class VersionCollectionTest extends GroovyTestCase {
    * branched from Major-1.x At the time of this writing 6.2 is unreleased and 6.3 is the 6.x branch. This test simulates the behavior
    * from 7.0 perspective, or master at the time of this writing.
    */
+  @Test
```

Comment:
I think this will be confusing to start having @Test in gradle tests, when we do not have them in the rest of the tests (and it is unclear how this would work when we run this as a normal `elasticsearch.build` as the `build-tools` project that uses randomized runner). We should be able to annotate the test with a junit3 compatible selector like randomized runner does?

Reply:
We could inherit from TestCase but we loose `@Rule` and probably `@Before`, `@After` etc. 
The rest of the code supports both `@Test` and methods starting with `test`, but this is implemented in `@RunWith(RandomizedRunner.class)`, `@TestMethodProviders` is _not_ a junit4 annotation, the junit3 compatible selector is part of lucene test, so while it's possible to bring that in as a dependency it seems a bit odd. The other possibility is to duplicate that one filter class.

Reply:
I would duplicate the class if necessary. Having @Test in some parts of our code would be very confusing.

Reply:
done